src/utils/location.test.js

```diff
@@ -1,3 +1,6 @@
+// TODO(pablo): The type for loc is really verbose below.  Gotta be a
```

Comment:
`Location` by default, is a type that is sourced from the DOM. That type is immutable and does not allow you to create it as it has no constructor.

If we were in pure TypeScript mode, we could use a type that implements the DOM's interface. Instead, in my draft PR, I create an object that mimics the members' types and a function to return new instances of it.

Then within the tests I manipulate the Location-like object for my test subjects.

https://github.com/bldrs-ai/Share/pull/504/files#diff-11e2a89e0bb7478d9e0add3e59943227c24b3d70c93ec5339007422ff993d335

Reply:
Done.. used chatgpt saved me on the typedef!